sim/CMakeLists.txt

```diff
@@ -14,6 +23,13 @@ add_executable(${PROJECT_NAME} ${SOURCES})
 add_executable("${PROJECT_NAME}_test" ${TESTS})
 target_include_directories(${PROJECT_NAME} PRIVATE ${SRC_DIR})
 
+# Link libraries
+foreach (lib IN ITEMS util ics)
+    add_subdirectory("${SRC_DIR}/${lib}")
+    target_link_libraries(${PROJECT_NAME} PRIVATE ${lib})
+    target_link_libraries(${PROJECT_NAME}_test PRIVATE -Wl,--whole-archive ${lib}-test -Wl,--no-whole-archive)
+endforeach()
```

Comment:
I don't our project is at a stage where we should have multiple cmake lists. one major downside is that i can no longer run all the tests at once.